clients/src/main/java/org/apache/kafka/common/security/ssl/SslFactory.java

```diff
@@ -99,7 +100,7 @@ public void configure(Map<String, ?> configs) throws KafkaException {
             try {
                 SslEngineValidator.validate(builder, builder);
             } catch (Exception e) {
-                throw new ConfigException("A client SSLEngine created with the provided settings " +
+                throw new InvalidConfigurationException("A client SSLEngine created with the provided settings " +
```

Comment:
Usages here are a bit tricky to track. So many overrided "configure" methods that go in circles 😅 
@bob-barrett do you have a simplified flow for this to confirm it won't cause compatibility issues?